File: algebra/factorial/javascript/factorial.js
// ES5

function factorial(n) {
	var result = 1;
	
	for(var i = 1; i <= n; i++){
		result = result * i;
	}

	return result;
}

// ES6

let factorial = (n) => {
	let result = 1;
	
	for(let i = 1; i <= n; i++){
		result = result * i;
	}

	return result;
}

File: algebra/factorial/golang/factorial.go
// Computes the factorial of a number

package main

import "fmt"

func factorial(x int64) int64{
	if (x==0 || x==1){
		return 1
	} else{
		result := x * factorial(x-1)
		return result
	}
}

func main(){
	fmt.Print("The factorial of 5 is ")
	fmt.Println(factorial(5))
}